app/services/macros/execution_service.rb

```diff
@@ -22,6 +22,16 @@ def perform
 
   private
 
+  def add_private_note(message)
+    return if conversation_a_tweet?
+
+    params = { content: message[0], private: true }
+
+    # Added reload here to ensure conversation us persistent with the latest updates
+    mb = Messages::MessageBuilder.new(@user, @conversation.reload, params)
+    mb.perform
+  end
```

Comment:
@tejaswinichile Couldn't this be a common function for both send_message and private note?